### PR TITLE
docs: update live demo logic with dumi 2.3 beta

### PR DESCRIPTION
**Pull request by PeachScript**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

#45383

### 💡 Background and solution

Migrate to the updated live demo api since dumi 2.3.0-beta.0.

Improvements:
1. Lazy load react-dom/server and sucrase for live demo
2. Use textarea instead of react-editor to reduce bundle size
3. Live demo also support iframe mode
4. Live demo API is simpler
5. Expand clickable area for source copy button

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | --        |
| 🇨🇳 Chinese | --        |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -227,7 +227,7 @@
     "cross-fetch": "^4.0.0",
     "crypto": "^1.0.1",
     "dekko": "^0.2.1",
-    "dumi": "2.3.0-alpha.9",
+    "dumi": "^2.3.0-beta.0",
```
Comment (yoyo837) [inline package.json:230]: 下面还有个overrides需要改吗？

Comment (MadCcc) [reply package.json]: 不用改，那个可以等到发布正式版的时候删掉

Comment (PeachScript) [reply package.json]: 这个 overrides 感觉没啥用，我删了试试看

Comment (yoyo837) [reply package.json]: 那个插件会匹配到2.2.0的正式版，高版本的npm， install的时候需要 --force

Comment (PeachScript) [reply package.json]: 好的，那我一起调下

**Review by MadCcc — APPROVED** [review]

**Review by li-jia-nan — APPROVED** [review]

**Comment by PeachScript** [comment]
@MadCcc 重新部署了，有空再看看有没有其他问题

**Pull request merged**